github/repos.go

```diff
@@ -579,6 +581,9 @@ type PullRequestReviewsEnforcementRequest struct {
 	DismissStaleReviews bool `json:"dismiss_stale_reviews"`
 	// RequireCodeOwnerReviews specifies if an approved review is required in pull requests including files with a designated code owner.
 	RequireCodeOwnerReviews bool `json:"require_code_owner_reviews"`
+	// RequiredApprovingReviewCount specifies the number of approvals required before the pull request can be merged.
+	// Can be omitted.
+	RequiredApprovingReviewCount *int `json:"required_approving_review_count,omitempty"`
```

Comment:
This parameter appears to be mandatory, not optional, here.

https://developer.github.com/v3/repos/branches/#update-branch-protection lists `required_approving_review_count` in a table labeled as:

> The `required_pull_request_reviews` object **must** have the following keys

Emphasis on _must_.

So, it should be:

```Go
RequiredApprovingReviewCount int `json:"required_approving_review_count"`
```

Reply:
That was my initial thought as well, according to the docs its required. That said, the request has worked thus far without having it?

Reply:
The documentation at https://developer.github.com/v3/repos/branches/#update-branch-protection has this note (which links to https://developer.github.com/changes/2018-03-16-protected-branches-required-approving-reviews/):

![image](https://user-images.githubusercontent.com/1924134/38158582-673fbf84-3465-11e8-9e5e-d3b32792eaa8.png)

It doesn't look like we're setting that `application/vnd.github.luke-cage-preview+json` custom Accept header, are we?